cosmo_tester/test_suites/snapshots/single_tenant_snapshot_test.py

```diff
@@ -0,0 +1,145 @@
+########
+# Copyright (c) 2017 GigaSpaces Technologies Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+#    * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    * See the License for the specific language governing permissions and
+#    * limitations under the License.
+
+import pytest
+
+from . import (
+    assert_hello_worlds,
+    check_deployments,
+    check_from_source_plugin,
+    check_plugins,
+    _cluster,
+    confirm_manager_empty,
+    create_noinstall_helloworld,
+    create_snapshot,
+    delete_manager,
+    download_snapshot,
+    get_deployments_list,
+    get_nodes,
+    get_secrets_list,
+    NOINSTALL_DEPLOYMENT_ID,
+    remove_and_check_deployments,
+    restore_snapshot,
+    SNAPSHOT_ID,
+    upgrade_agents,
+    upload_and_install_helloworld,
+    upload_snapshot,
+    upload_test_plugin,
+)
+
+
+def test_restore_snapshot_and_agents_upgrade_singletenant(
+        cfy, cluster_singletenant, attributes, logger, tmpdir):
+    local_snapshot_path = str(tmpdir / 'snapshot.zip')
+    old_manager = cluster_singletenant.managers[0]
+    new_manager = cluster_singletenant.managers[1]
+    hello_vm = cluster_singletenant.managers[2]
+
+    confirm_manager_empty(new_manager)
+
+    upload_and_install_helloworld(attributes, logger, old_manager,
+                                  hello_vm, tmpdir)
+    create_noinstall_helloworld(old_manager, logger)
+
+    old_plugins = upload_test_plugin(old_manager, logger)
+    old_deployments = get_deployments_list(old_manager)
+
+    create_snapshot(old_manager, SNAPSHOT_ID, attributes, logger)
+    download_snapshot(old_manager, local_snapshot_path, SNAPSHOT_ID, logger)
+    upload_snapshot(new_manager, local_snapshot_path, SNAPSHOT_ID, logger)
+
+    restore_snapshot(new_manager, SNAPSHOT_ID, cfy, logger)
+
+    assert_hello_worlds([hello_vm], installed=True, logger=logger)
+
+    check_secrets_converted(new_manager, logger)
+    check_plugins(new_manager, old_plugins, logger)
+    check_deployments(new_manager, old_deployments, logger)
+    check_from_source_plugin(
+        new_manager,
+        'aws',
+        NOINSTALL_DEPLOYMENT_ID,
+        logger,
+    )
+
+    upgrade_agents(cfy, new_manager, logger)
+
+    # The old manager needs to exist until the agents install is run
+    delete_manager(old_manager, logger)
+
+    remove_and_check_deployments([hello_vm], new_manager, logger)
+
+
+@pytest.fixture(
+        scope='module',
+        params=['4.0', '3.4.2'])
+def cluster_singletenant(request, cfy, ssh_key, module_tmpdir, attributes,
```

Comment:
They're gradually becoming not that way wherever I'm changing them, so that when one first looks at the file one sees the tests rather than a nest of random fixtures, functions, and other garbage with tests nestled in the middle.